#! /bin/sh -e

# DP: Add support for db4.6.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- setup.py~	2008-05-05 22:21:38.000000000 +0200
+++ setup.py	2008-05-23 13:04:00.000000000 +0200
@@ -685,7 +685,7 @@
         # a release.  Most open source OSes come with one or more
         # versions of BerkeleyDB already installed.
 
-        max_db_ver = (4, 5)  # XXX(gregory.p.smith): 4.6 "works" but seems to
+        max_db_ver = (4, 6)  # XXX(gregory.p.smith): 4.6 "works" but seems to
                              # have issues on many platforms.  I've temporarily
                              # disabled 4.6 to see what the odd platform
                              # buildbots say.
